CHANGELOG.md

```diff
@@ -14,6 +14,90 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - For MultiGroupServer, the backend and frontend share each node
 - Moved `MAPL_HistoryGridComp.F90` and related files to `gridcomps/History`
 
+- Major refactoring related to MAPL generic capabilities
+
+    0. Cretaed new subdirectory  "generic"
+    
+    1. Modified interface to MAPL_Get() to use ALLOCATABLE for GEX, GIM,
+       GCS, and GCnamelist.  Original interface assumed these are
+       contiguous which will not be sustainable under the planned changes.
+    
+    2. (Re)Introduce fundamental classes in generic subdir.  Made
+       MAPL_MetaComp an extension of AbstractComponentNode, and provided
+       stub implementations for 3 methods. (Commenting out other
+       interfaces in the abstract class for now.)
+    
+    3. Activate get_parent(), add_child(), and num_children()
+    
+    4. Introduced MaplComponentNode
+    
+       Easier to introduce things in a smaller module.   Replaced String NodeMap with
+       an internal map in MaplComponentNdoed wcich uses pointers.   Allows use to
```

Comment:
Spelling: "MaplComponentNdoed" Not sure what this should be...

Reply:
Yeah - this comment is already out of date.   Closest analog now is ConcreteComposite and StringCompositeMap. Will fix.